vmdb/app/models/ems_kubernetes.rb

```diff
@@ -0,0 +1,21 @@
+class EmsKubernetes < EmsContainer
+  has_many :container_nodes,                      :foreign_key => :ems_id, :dependent => :destroy
+  has_many :container_groups,                       :foreign_key => :ems_id, :dependent => :destroy
+  has_many :container_services,                   :foreign_key => :ems_id, :dependent => :destroy
+
+  def self.ems_type
+    @ems_type ||= "kubernetes".freeze
+  end
+
+  def self.description
+    @description ||= "Kubernetes".freeze
+  end
+
+  def inventory
+    require 'kubeclient'
+    # TODO, timeout params
+    url = "http://#{hostname}:#{port}/api/"
+    client = Kubeclient::Client.new  url, "v1beta3"
+    client.get_all_entities
+  end
```

Comment:
kubeclient is written by @abonas, so that's convenient :stuck_out_tongue: .  @abonas, just be sure kubeclient can accept a URI object (i.e. calls .to_s on whatever it gets).